File: Java/cpf.java
import java.util.regex.*;

public class ValidadorCPF {

    public static boolean validarCPF(String cpf) {
        // Remove tudo que não for número
        cpf = cpf.replaceAll("[^0-9]", "");

        // Verifica se tem 11 dígitos
        if (cpf.length() != 11) {
            return false;
        }

        // Verifica se todos os dígitos são iguais
        if (cpf.matches("(\\d)\\1{10}")) {
            return false;
        }

        // Validação do primeiro dígito
        int soma = 0;
        for (int i = 0; i < 9; i++) {
            soma += Character.getNumericValue(cpf.charAt(i)) * (10 - i);
        }
        int digito1 = (soma * 10) % 11;
        if (digito1 == 10 || digito1 == 11) {
            digito1 = 0;
        }
        if (digito1 != Character.getNumericValue(cpf.charAt(9))) {
            return false;
        }

        // Validação do segundo dígito
        soma = 0;
        for (int i = 0; i < 10; i++) {
            soma += Character.getNumericValue(cpf.charAt(i)) * (11 - i);
        }
        int digito2 = (soma * 10) % 11;
        if (digito2 == 10 || digito2 == 11) {
            digito2 = 0;
        }
        if (digito2 != Character.getNumericValue(cpf.charAt(10))) {
            return false;
        }

        return true;
    }

    public static void main(String[] args) {
        // Exemplo de uso
        String[] cpfsParaTestar = {
            "123.456.789-09",  // Válido
            "111.111.111-11",  // Inválido
        };

        for (String cpf : cpfsParaTestar) {
            System.out.println("CPF " + cpf + " é válido? " + (validarCPF(cpf) ? "Sim" : "Não"));
        }
    }
}

File: Python/cpf.py
import re

def validar_cpf(cpf: str) -> bool:
    # Remove caracteres não numéricos
    cpf = re.sub(r'[^0-9]', '', cpf)
    
    # Verifica se o CPF tem 11 dígitos
    if len(cpf) != 11:
        return False
    
    # Verifica se todos os dígitos são iguais (ex: 111.111.111-11 é inválido)
    if cpf == cpf[0] * 11:
        return False

    # Valida o primeiro dígito
    soma = sum(int(cpf[i]) * (10 - i) for i in range(9))
    digito1 = (soma * 10 % 11) % 10
    if digito1 != int(cpf[9]):
        return False

    # Valida o segundo dígito
    soma = sum(int(cpf[i]) * (11 - i) for i in range(10))
    digito2 = (soma * 10 % 11) % 10
    if digito2 != int(cpf[10]):
        return False

    return True

# Exemplo de uso:
# Exemplo de uso
cpfs_para_testar = [
    "123.456.789-09",  # Válido
    "111.111.111-11",     # Inválido
]

for cpf in cpfs_para_testar:
    print(f"CPF {cpf} é válido? {validar_cpf(cpf)}")